src/request_queues.js

```diff
@@ -0,0 +1,340 @@
+     * @param {Object} options
+     * @param options.token
+     * @param {Number} [options.offset=0] - Number of array elements that should be skipped at the start.
+     * @param {Number} [options.limit=1000] - Maximum number of array elements to return.
+     * @param {Number} [options.desc] - If 1 then the objects are sorted by the startedAt field in descending order.
+     * @param callback
+     * @returns {PaginationList}
+     */
+    listQueues: (requestPromise, options) => {
+        const { baseUrl, token, offset, limit, desc, unnamed } = options;
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(limit, 'limit', 'Maybe Number');
+        checkParamOrThrow(offset, 'offset', 'Maybe Number');
+        checkParamOrThrow(desc, 'desc', 'Maybe Boolean');
+        checkParamOrThrow(unnamed, 'unnamed', 'Maybe Boolean');
+
+        const query = { token };
+
+        if (limit) query.limit = limit;
+        if (offset) query.offset = offset;
+        if (desc) query.desc = 1;
+        if (unnamed) query.unnamed = 1;
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}`,
+            json: true,
+            method: 'GET',
+            qs: query,
+        })
+            .then(pluckData);
+    },
+
+    /**
+     * Gets request queue.
+     *
+     * @memberof ApifyClient.requestQueues
+     * @instance
+     * @param {Object} options
+     * @param {String} options.queueId - Unique queue ID
+     * @param callback
+     * @returns {RequestQueue}
+     */
+    getQueue: (requestPromise, options) => {
+        const { baseUrl, queueId } = options;
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(queueId, 'queueId', 'String');
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}/${queueId}`,
+            json: true,
+            method: 'GET',
+        })
+            .then(pluckData)
+            .catch(catchNotFoundOrThrow);
+    },
+
+    /**
+     * Deletes request queue.
+     *
+     * @memberof ApifyClient.requestQueues
+     * @instance
+     * @param {Object} options
+     * @param {String} options.queueId - Unique queue ID
+     * @param callback
+     * @returns {*}
+     */
+    deleteQueue: (requestPromise, options) => {
+        const { baseUrl, queueId } = options;
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(queueId, 'queueId', 'String');
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}/${queueId}`,
+            json: true,
+            method: 'DELETE',
+        });
+    },
+
+    /**
+     * @typedef {Object} AddedRequetInfo
+     * @property {Boolean} wasAlreadyPresent Indicates if request was already present in the queue.
+     * @property {Boolean} wasAlreadyHandled Indicates if request was already marked as handled.
+     * @property {String} id The ID of the added request
```

Comment:
it would be better to call this "requestId" than just simply "id", to make it more clear